### PR TITLE
Add toggles for heightfield filtering

### PR DESCRIPTION
**Pull request by twoscomplement**
I wanted to see the effect of the various filtering options available, so I added toggles for each of them.

### CODE REVIEW

**Comment by twoscomplement** [comment]
OK, I've added that.

File: RecastDemo/Source/Sample.cpp [hunk 1]
```diff
@@ -42,7 +42,10 @@ Sample::Sample() :
 	m_crowd(0),
 	m_navMeshDrawFlags(DU_DRAWNAVMESH_OFFMESHCONS|DU_DRAWNAVMESH_CLOSEDLIST),
 	m_tool(0),
-	m_ctx(0)
+	m_ctx(0),
+	m_filterLowHangingObstacles(true),
```
Comment (hymerman) [inline RecastDemo/Source/Sample.cpp]: Nitpick: these should go between `m_navMeshDrawFlags` and `m_tool` to match member ordering in the class.

**Comment by twoscomplement** [comment]
How about you add -Werror=reorder so this kind of thing is caught by the automated build, and I'll make that change.

**Comment by hymerman** [comment]
Weird, I'm trying to, but can't get it to work locally. I added the flag to buildoptions in the Premake config but it's ignoring it. I can set it in CXXFLAGS in the Travis build script but I'd rather have it as part of the Premake config so it affects users before pushing. I'll try setting the 'FatalWarnings' flag for just Linux, as there are tons of warnings on other platforms right now - there are still a few warnings to fix up before pushing that though.

I figure that having warnings as errors on one platform is a good rabbit hole to have fallen down ;)

**Comment by hymerman** [comment]
Done! Just need #245 and #248 to be merged, then this will start failing the Travis build.

**Comment by hymerman** [comment]
Done - it doesn't look like there's a way to tell this PR to re-run the checks 'as if' it were already merged into master though, so they won't fail yet. Could you rebase and make the change please? I'd do it myself but I think it'll take the credit away from you :)

**Comment by jakobbotsch** [comment]
Looks like closing and reopening worked.

**Comment by hymerman** [comment]
Oh cool! You learn something new every day :) Thanks Jakob!

**Comment by twoscomplement** [comment]
Nice! Thanks for making that change to the build.

I've changed the order of the members in the initialization list.

**Comment by hymerman** [comment]
Looking good! Thanks again :)

**Pull request merged**